src/Compilers/CSharp/Portable/Symbols/NativeIntegerTypeSymbol.cs

```diff
@@ -0,0 +1,67 @@
+{
+    internal sealed class NativeIntegerTypeSymbol : WrappedNamedTypeSymbol
+    {
+        internal NativeIntegerTypeSymbol(NamedTypeSymbol underlying) : base(underlying)
+        {
+            Debug.Assert(!underlying.IsNativeInt);
+            Debug.Assert(underlying.SpecialType == SpecialType.System_IntPtr || underlying.SpecialType == SpecialType.System_UIntPtr);
+            Debug.Assert(this.Equals(underlying));
+            Debug.Assert(underlying.Equals(this));
+        }
+
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+
+        public override NamedTypeSymbol ConstructedFrom => _underlyingType.ConstructedFrom;
+
+        public override Symbol ContainingSymbol => _underlyingType.ContainingSymbol;
+
+        internal override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotationsNoUseSiteDiagnostics => ImmutableArray<TypeWithAnnotations>.Empty;
+
+        internal override bool IsComImport => _underlyingType.IsComImport;
+
+        internal override NamedTypeSymbol BaseTypeNoUseSiteDiagnostics => _underlyingType.BaseTypeNoUseSiteDiagnostics;
+
+        public override SpecialType SpecialType => _underlyingType.SpecialType;
+
+        public override IEnumerable<string> MemberNames => throw new NotImplementedException();
+
+        public override ImmutableArray<Symbol> GetMembers() => _underlyingType.GetMembers();
+
+        public override ImmutableArray<Symbol> GetMembers(string name) => _underlyingType.GetMembers(name);
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers() => _underlyingType.GetTypeMembers();
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name) => _underlyingType.GetTypeMembers(name);
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int arity) => _underlyingType.GetTypeMembers(name, arity);
+
+        internal override NamedTypeSymbol GetDeclaredBaseType(ConsList<TypeSymbol> basesBeingResolved) => _underlyingType.GetDeclaredBaseType(basesBeingResolved);
+
+        internal override ImmutableArray<NamedTypeSymbol> GetDeclaredInterfaces(ConsList<TypeSymbol> basesBeingResolved) => _underlyingType.GetDeclaredInterfaces(basesBeingResolved);
+
+        internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers() => throw ExceptionUtilities.Unreachable;
+
+        internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers(string name) => throw ExceptionUtilities.Unreachable;
+
+        internal override IEnumerable<FieldSymbol> GetFieldsToEmit() => throw ExceptionUtilities.Unreachable;
+
+        internal override ImmutableArray<NamedTypeSymbol> GetInterfacesToEmit() => throw ExceptionUtilities.Unreachable;
+
+        internal override ImmutableArray<NamedTypeSymbol> InterfacesNoUseSiteDiagnostics(ConsList<TypeSymbol> basesBeingResolved = null) => _underlyingType.InterfacesNoUseSiteDiagnostics(basesBeingResolved);
```

Comment:
>_underlyingType.InterfacesNoUseSiteDiagnostics(basesBeingResolved) [](start = 140, length = 66)

Do we want to substitute reference to itself, if any?

Reply:
Added a PROTOTYPE comment for now.

---
In reply to: [370296295](https://github.com/dotnet/roslyn/pull/40994#discussion_r370296295) [](ancestors = 370296295)